argoverse/data_loading/argoverse_tracking_loader.py

```diff
@@ -171,20 +181,66 @@ def image_list_sync(self) -> Dict[str, List[np.ndarray]]:
                         List[int],
                         list(
-                            filter(
-                                lambda x: x is not None,
-                                (
-                                    self.sync.get_closest_cam_channel_timestamp(x, camera, log)
-                                    for x in self._lidar_timestamp_list[log]
-                                ),
-                            )
+                            self.sync.get_closest_cam_channel_timestamp(x, camera, log)
+                            for x in self._lidar_timestamp_list[log]
                         ),
                     )
 
-                    self._image_list_sync[log][camera] = [
-                        self.get_image_at_timestamp(x, camera=camera, log_id=log, load=False)
-                        for x in self._image_timestamp_list_sync[log][camera]
-                    ]
+                    assert len(self._image_timestamp_list_sync[log][camera]) == len(
+                        self._lidar_timestamp_list[log]
+                    ), "Missing timestamp"
+
+                    self._image_list_sync[log][camera] = []
+                    for i, x in enumerate(self._image_timestamp_list_sync[log][camera]):
+                        closest_gap_backward = None
+                        closest_ts_backward = None
+                        closest_gap_forward = None
+                        closest_ts_forward = None
+                        if x is None:
+                            # missing timestamp
+                            new_timestamp = None
+                            if i != 0:
+                                for ii in range(i - 1, -1, -1):  # loop backward
+                                    x = self._image_timestamp_list_sync[log][camera][ii]
+                                    if x is not None:
+                                        closest_gap_backward = self._lidar_timestamp_list[log][i] - x
+                                        assert closest_gap_backward > 0
+                                        closest_ts_backward = x
+                                        break
+                            if i != len(self._image_timestamp_list_sync[log][camera]) - 1:
+                                for ii in range(
+                                    i + 1, len(self._image_timestamp_list_sync[log][camera])
+                                ):  # loop forward
+                                    x = self._image_timestamp_list_sync[log][camera][ii]
+                                    if x is not None:
+                                        closest_gap_forward = x - self._lidar_timestamp_list[log][i]
+                                        assert closest_gap_forward > 0
+                                        closest_ts_forward = x
+                                        break
+                            if self.return_missing_image_as == "closest":
```

Comment:
i think a cleaner way to do this is to just call `get_closest_cam_channel_timestamp`, but with a flag that states `ignore_ts_threshold=True`. It would be a new default argument set to False usually
https://github.com/argoai/argoverse-api/blob/master/argoverse/data_loading/synchronization_database.py#L172

Then I think we can eliminate 15 lines of code here

Reply:
we would need to change `get_closest_cam_channel_timestamp()` so that if
```python
if ignore_ts_threshold == True:
    return closest_cam_ch_timestamp
```
immediately, instead of doing the timestamp diff checks